analyzers/nodejs/integration_test.go

```diff
@@ -102,59 +109,46 @@ func assertProjectFixtureExists(t *testing.T, name string) {
 	assert.True(t, exists, name+" did not have its node modules installed")
 }
 
-func initializeProjects(testDir string) error {
-	var waitGroup sync.WaitGroup
-	waitGroup.Add(len(projects))
-
-	for _, project := range projects {
-		go func(proj fixtures.Project) {
-			defer waitGroup.Done()
-
-			projectDir := filepath.Join(testDir, proj.Name)
-			log.Debug("initializing " + projectDir)
-			nodeModulesExist, err := files.ExistsFolder(projectDir, "node_modules")
-			if err != nil {
-				panic(err)
-			}
+func projectInitializer(proj fixtures.Project, projectDir string) error {
+	nodeModulesExist, err := files.ExistsFolder(projectDir, "node_modules")
+	if err != nil {
+		return err
+	}
 
-			if nodeModulesExist {
-				log.Debug("node modules already exists for " + proj.Name + "skipping initialization")
-				return
-			}
+	if nodeModulesExist {
+		log.Debug("node modules already exists for " + proj.Name + "skipping initialization")
```

Comment:
I'd be more comfortable with this if we validated `node_modules` in some way, but I don't think that's worth implementing. A test that passes due to a bad cache would probably fail in CI anyway (unless the CI cache also got polluted -- how likely is that?).

Reply:
I don't think it's likely CI will get polluted. We update the cache every time we add a new integration test project fixture (and whenever a particular project initialization needs an update).